recommendations/views.py

```diff
@@ -15,52 +21,60 @@ def recommend_view(request):
 
 
 # narrow the possible recommendations by querying based on submitted forms
-# TODO more than just genres (generic problems)
 def narrow_view(request):
-    form = GenreForm(request.POST)  # TODO generic
+    if request.user.is_authenticated is False:
+        form = FORMS[random.randint(0, len(FORMS) - 1)]  # guests get to test
+        recommendation = {"possible_film_count": 27122, "step": 1}
+        return render(
+            request,
+            "recommendations/index.html",
+            {"form": form, "recommendation": recommendation},
+        )
 
-    # keep track of for logged in user, give a random joe any ol' form next (experience w/o actual rec)
-    if form.is_valid() and request.user.is_authenticated:
-        user = User.objects.get(username=request.user)
-        recommendation = Recommendation.objects.get(user_id=user)
-        step = recommendation.step
-        selection = form.cleaned_data.get(FIELD[step], [])
-
-        # first question will always be genre - populate possible films for the first time
-        if step == 0:
-            movies = Movie.objects.filter(genres__contains=selection)
-        else:
-            movies = recommendation.possible_films.filter(
-                genres__contains=selection
-            )  # TODO generic?
+    user = User.objects.get(username=request.user)
+    recommendation = Recommendation.objects.get(user_id=user)
+    step = recommendation.step
+
+    form = FORMS[step](request.POST)
+    if form.is_valid():
+        field = FIELD[step]
+        selection = form.cleaned_data.get(field, [])
 
+        movies = recommendation_querying(
+            recommendation, MOVIE_MODEL_COMPLEMENT[step], selection
+        )
         recommendation.possible_films.set(movies)
```

Comment:
I don't think this line is actually updating the possible_films field? My recommendation count after doing action-comedies from 2000 and after does have the count decrease to 485 but checking via the admin site shows all movies (or at the very least more than 485, and also a movie that didn't have action/comedy in it's genres was present)

Reply:
it does
only the highlighted movies are the ones included... it's very odd, but yes, it works.

Reply:
Oh wow I see now, very strange design